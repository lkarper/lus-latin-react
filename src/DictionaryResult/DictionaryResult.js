import React, { Component } from 'react';

class DictionaryResult extends Component {
    render() {
        const { data } = this.props;
        
        let resultsHTML;

        if (!Object.keys(data).length) {
            resultsHTML = <p>Enter a word above to search the dictionary.</p>;
        } else {
            if (!data.match) {
                resultsHTML = <p>Sorry, no results found.  Please check your input and try again.</p>;
            } else {
                resultsHTML = (
                <ol> 
                    {data.results.map(entry => {
                        const { declension, gender, key, main_notes, part_of_speech, senses, title_genitive, title_orthography } = entry;
                        return (
                            <li key={key}>
                                <h3>{key}</h3>
                                <p>{title_genitive}</p>
                                <p>{title_orthography}</p>
                                <p>{gender}</p>
                                <p>{part_of_speech}</p>
                                <p>{declension}</p>
                                <p>{main_notes}</p>
                                <ul>{senses.map((sense, i) => <li key={i}>{sense}</li>)}</ul>
                            </li>
                        ); 
                    })}
                </ol>)
            }
        }

        return (
            <div>
                {resultsHTML}
            </div>
        );
    }
}

export default DictionaryResult;