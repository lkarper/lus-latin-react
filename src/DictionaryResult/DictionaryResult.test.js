import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DictionaryResult from './DictionaryResult';

describe('DictionaryResult component', () => {
    it('renders the component as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DictionaryResult />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the UI as expected when no data is passed in props', () => {
        const wrapper = shallow(<DictionaryResult />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});