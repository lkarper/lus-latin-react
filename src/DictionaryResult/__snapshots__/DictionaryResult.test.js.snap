// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictionaryResult component renders the UI as expected when no data is passed in props 1`] = `
<div>
  <p>
    Enter a word above to search the dictionary.
  </p>
</div>
`;

exports[`DictionaryResult component renders the UI as expected when no match is found 1`] = `
<div>
  <p>
    Sorry, no results found.  Please check your input and try again.
  </p>
</div>
`;
